### PR TITLE
enhancements on scicrunch classifiers (backend)

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Some enhancements and increases test coverage on scicrunch classifiers

### Fixes error responses
- Possible outcomes from ``POST /groups/sparc/classifiers/scicrunch-resources/{rrid}``
- **IMPORTANT**: ``rrid`` should NOT have prefix ``RRID:``
```yaml
responses:
        "200":
          description: Got information of a valid RRID
        "400":
          description: Invalid RRID
        "503":
          description: scircrunch.org service is not reachable
```

### non "SCR_" RRIDs
- scicrunch.org API only works with RRIDs defined in scicrunch registry tools (prefixed with ``SCR_``)
- The rest are all treated as invalid identifiers even if they can be searched via their website 
- See trials and classification done in ``services/web/server/tests/unit/isolated/test_scicrunch_service_api.py``

## Related issue/s

- ITISFoundation/osparc-issues#344: J-D4.6. A computational Minimal Information Standard (cMIS) has been established Y4M03

## How to test

Get a token in scicrunch.org for API
```cmd
export SCICRUNCH_API_KEY=some-secret-token
make devenv
cd services/web/server
make install-dev
make test-dev-unit
```

## Checklist

- [x] @GitHK reviews from #2045
- [x] Fix: if auth error (e.g. invalid token) now it is returning notfound!
- [x] Check queries with non SCR_ RRIDs
- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/web/server/tests/unit/with_dbs/fast/test_groups_handlers_classifers.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        cfg[section]["enabled"] = False
+
+    # NOTE: To see logs, use pytest -s --log-cli-level=DEBUG
+    ## setup_logging(level=logging.DEBUG)
+
+    return cfg
+
+
+@pytest.mark.skip(reason="UNDER DEV: test_group_handlers")
+async def test_unauntheticated_request_to_scicrunch(client):
```
Comment (sanderegg) [inline services/web/server/tests/unit/with_dbs/fast/test_groups_handlers_classifers.py:64]: weird name

**Review by sanderegg — APPROVED** [review]
:clap: :tophat:

File: services/web/server/tests/unit/with_dbs/fast/test_groups_handlers_classifers.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/tests/unit/with_dbs/fast/test_groups_handlers_classifers.py:64]: :-D

**Review by odeimaiz — APPROVED** [review]
💪

**Review by mguidon — APPROVED** [review]
Go ahead.

**Pull request merged**